src/components/UIElements/BasketDashboardHeader.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { useEffect } from 'react';
+import BasketBox from './BasketBox';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Switch from '@mui/material/Switch';
+import { useTheme } from '@mui/material/styles';
+import { useSelector, useDispatch } from 'react-redux';
+import StyledMenu from './StyledMenu';
+import OutlinedInput from '@mui/material/OutlinedInput';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import Divider from '@mui/material/Divider';
+import { changePage, changeType, changeOrder } from '../../redux/basket/basketSlice'
+
+import { get_owner_baskets, get_coowner_baskets, get_public_baskets, get_private_baskets } from '../../redux/basket/basketActions';
+
+const ITEM_HEIGHT = 50;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 5.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
+
+const buttonOptions = [
+  "Created by me",
+  "Coowned by me",
+  "Public baskets",
+  "Private baskets"
+]
+
+const orderOptions = [
+  "Newest to oldest",
+  "Oldest to newest",
+  "Divider",
+  "Expensive to cheap",
+  "Cheap to expensive",
+  "Divider",
+  "Soon to expire",
+  "Far to expire"
+]
+
+const BasketDashboardHeader = () => {
+  const { page } = useSelector(state => state.basket.paginationData);
+
+  const [hiddenFilterAnchor, sethiddenFilterAnchor] = React.useState(null);
```

Comment:
as you import useEffect directly from react, not React.useEffect, then do the same thing for use state